airbyte-integrations/connectors/source-posthog/source_posthog/streams.py

```diff
@@ -0,0 +1,312 @@
+            last_stream_curfield = stream_state[self.cursor_field]
+            if last_stream_curfield >= last_record_curfield:
+                yield from []
+            else:
+                first_record_curfield = data[0][self.cursor_field]
+                if first_record_curfield > last_stream_curfield:
+                    yield from data
+                else:
+                    slice_ = [i for i in data if i[self.cursor_field] > last_stream_curfield]
+                    yield from slice_
+
+        else:
+            yield from data
+
+
+class Elements(PosthogStream):
+    data_field = "elements"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "element/stats"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        elements = [i[self.data_field][0] for i in response_json]
+        yield from elements
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This enpoint returns full stats list and have no pagination.
+        """
+        return None
+
+
+class Events(IncrementalPosthogStream):
+    """
+    CAUTION! API supports inserting custom timestamp.
+    That may break incremental since there is no way to sort in request params.
+    """
+
+    cursor_field = "timestamp"
+    data_field = "results"
+
+    def __init__(self, start_date: str, **kwargs):
+        self.start_date = start_date
+        super().__init__(**kwargs)
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "event"
+
+    def request_params(self, stream_state=None, stream_slice: Mapping[str, Any] = None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        since_value = stream_state.get(self.cursor_field) or self.start_date
+        since_value = max(since_value, self.start_date)
+        if since_value:
+            params["after"] = since_value
+        return params
+
+
+class EventsSessions(PosthogStream):
+    data_field = "result"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "event/sessions"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        resp_json = response.json()
+        if "pagination" in resp_json and resp_json["pagination"]:
+            return resp_json["pagination"]
+
+
+class FeatureFlags(IncrementalPosthogStream):
+    cursor_field = "id"
+    data_field = "results"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return "feature_flag"
+
+
+class Insights(PosthogStream):
+    """
+    NO WAY TO SORT TO IMPLETEMENT INCREMENTAL! id, created_at, last_refresh
+    are ordered in a random (!) way, no DESC no ASC
+    """
+
+    data_field = "results"
```

Comment:
We very often set `data_field` as `results` or `result`. If we set one of these values as default in the base class, then we will reduce the number of variable overrides.